### ISSUE TITLE
 include the length of an array and the index in the error message 

### ISSUE THREAD

**Issue by Yongxuanzhang** [tektoncd/pipeline#6212]
currently the validation of array indexing will return nonexistent param references error. It would be better if we could include the length of an array and the index in the error message.

Background: https://github.com/tektoncd/pipeline/pull/6179#discussion_r1112399444

https://github.com/tektoncd/pipeline/blob/ac225de1b714b7a3cd78afe7155101e228789182/pkg/reconciler/pipelinerun/resources/validate_params.go#L131-L133

**Comment by afrittoli** [comment]
@Yongxuanzhang is this fixed now?
If you include `Fixes: #issue-number` in the commit message/PR description, the issue will automatically be closed once the PR is merged

**Comment by Yongxuanzhang** [comment]
> @Yongxuanzhang is this fixed now? If you include `Fixes: #issue-number` in the commit message/PR description, the issue will automatically be closed once the PR is merged

No it is not fixed, it was mentioned from a comment in the merged PR that we better have more explicit logging, this issue is in my queue with low priority but I will fix it
